File: frontend/chessandconnect/src/app/models/Games/Chess/ChessBasePiece.ts
import { ChessPieceColor } from "./Enums/Color"
import { Point } from "../Base/Point"
import { PieceType } from "./Enums/PieceType"

export interface ChessBasePiece {
    Id:number,
    Color: ChessPieceColor,
    Position: Point
    PieceType: PieceType
    PossibleMovements: Point[]
}

File: backend/chess4connect/chess4connect/Models/Games/Chess/Pieces/Base/ChessBasePiece.cs
using chess4connect.Models.Games.Base;
using chess4connect.Models.Games.Chess.Chess.Pieces.Types;
using System.Drawing;

namespace chess4connect.Models.Games.Chess.Chess.Pieces.Base;

public abstract class ChessBasePiece
{
    public int Id { get; set; }
    public bool Player1Piece { get; set; }
    public Point Position { get; set; }
    public PieceType PieceType { get; set; }
    public List<Point> BasicMovements { get; set; }
    public bool HasMoved { get; set; } = false;

    public ChessBasePiece(int id, PieceType pieceType, bool pieceColor, Point position)
    {
        PieceType = pieceType;
        Id = id;
        Player1Piece = pieceColor;
        Position = position;

        GetBasicMovements();
    }

    protected abstract void GetBasicMovements();
    

}
